tools/build_pak_files.py

```diff
@@ -0,0 +1,100 @@
+#
+# Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+#
+
+import argparse
+import glob
+import os
+import subprocess
+import sys
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--unreal_binaries_dir", required=True)
+    parser.add_argument("--unreal_project_dir", required=True)
+    parser.add_argument("--scene_id", required=True)
+    parser.add_argument("--platforms", nargs="*", required=True, help="Supported platform values are Windows, Linux, and Mac")
+    parser.add_argument("--output_dir", required=True)
+    args = parser.parse_args()
+
+    os.makedirs(args.output_dir, exist_ok=True)
+    
+    assert os.path.exists(args.unreal_binaries_dir)
+    unreal_editor_exe = os.path.realpath(os.path.join(args.unreal_binaries_dir, "Win64", "UE4Editor.exe"))
```

Comment:
This tool should be cross-platform. Why is building PAK files restricted to Windows?

Reply:
Updated to run on all supported platforms.

Upon testing on Linux, encountering an issue. I'm investigating it.

```bash
(spear) rachithp@rachithp:~/code/github/spear/tools (rachithp/build_paks)$ python build_pak_files.py --unreal_engine_dir /home/rachithp/code/github/unrealengine/UnrealEngine --contents_dir /media/rachithp/LaCie/spearsim_final_content_for_perforce --platforms Linux --output_dir /media/rachithp/LaCie/package_files_testing/linux --scene_names kujiale_0000
[SPEAR | build_pak_files.py] File or directory or symlink exists, removing: /home/rachithp/code/github/spear/cpp/unreal_projects/SpearSim/Content/Scenes/kujiale_0000
[SPEAR | build_pak_files.py] Creating symlink: /home/rachithp/code/github/spear/cpp/unreal_projects/SpearSim/Content/Scenes/kujiale_0000 -> /media/rachithp/LaCie/spearsim_final_content_for_perforce/Scenes/kujiale_0000
[SPEAR | build_pak_files.py] Executing: /home/rachithp/code/github/unrealengine/UnrealEngine/Engine/Binaries/Linux/UE4Editor /home/rachithp/code/github/spear/cpp/unreal_projects/SpearSim/SpearSim.uproject -run=Cook -TargetPlatform=LinuxNoEditor -fileopenlog -ddc=InstalledDerivedDataBackendGraph -unversioned -stdout -FullStdOutLogOutput -CrashForUAT -unattended -NoLogTimes -UTF8Output
/home/rachithp/code/github/unrealengine/UnrealEngine/Engine/Binaries/Linux/UE4Editor: symbol lookup error: /home/rachithp/code/github/unrealengine/UnrealEngine/Engine/Binaries/Linux/libUE4Editor-UnrealEd.so: undefined symbol: _ZN11FDDCCleanup8RunnableE
Traceback (most recent call last):
  File "/home/rachithp/code/github/spear/tools/build_pak_files.py", line 91, in <module>
    subprocess.run(cmd, check=True)
  File "/usr/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/home/rachithp/code/github/unrealengine/UnrealEngine/Engine/Binaries/Linux/UE4Editor', '/home/rachithp/code/github/spear/cpp/unreal_projects/SpearSim/SpearSim.uproject', '-run=Cook', '-TargetPlatform=LinuxNoEditor', '-fileopenlog', '-ddc=InstalledDerivedDataBackendGraph', '-unversioned', '-stdout', '-FullStdOutLogOutput', '-CrashForUAT', '-unattended', '-NoLogTimes', '-UTF8Output']' returned non-zero exit status 127.
```

Reply:
Confirmed that it was an issue with the UE on my system. I reinstalled UE 4.26 and now I'm able to execute this script successfully. The Linux pak file generated from the script also works with generate_image_dataset example.